components/ImageDisplay.vue

```diff
@@ -22,6 +28,19 @@ export default {
       this.$favorites.toggle(this.image).then(added => {
         this.$alert(added ? 'Added to favorites.' : 'Removed from favorites.');
       });
+    },
+    doubleClick() {
```

Comment:
This is a super overcomplicated solution compared to just registering a native dblclick event listener and binding it to the already existing favorite method.

Reply:
As far as I know the dblclick event listener doesn't work in mobile devices. Didn't know other way of doing that.